packages/core/primitives/event-dispatch/src/eventcontract.ts

```diff
@@ -282,9 +284,14 @@ export class EventContract implements UnrenamedEventContract {
     const earlyEventHandler: (event: Event) => void = earlyJsactionData.h;
     for (let idx = 0; idx < earlyEventTypes.length; idx++) {
       const eventType: string = earlyEventTypes[idx];
-      window.document.documentElement.removeEventListener(eventType, earlyEventHandler);
+      earlyJsactionData.c.removeEventListener(eventType, earlyEventHandler);
+    }
+    const earlyEventTypesCapture: string[] = earlyJsactionData.etc;
+    for (let idx = 0; idx < earlyEventTypesCapture.length; idx++) {
+      const eventType: string = earlyEventTypes[idx];
+      earlyJsactionData.c.removeEventListener(eventType, earlyEventHandler, /* useCapture */ true);
```

Comment:
nit: I think there might be an opportunity for refactoring - we can move the generic code to a utility function:
```suggestion
    const {c: container, et: eventTypes, etc: eventTypesWithCapture} = earlyJsactionData;
    removeEventListeners(container, eventTypes);
    removeEventListeners(container, eventTypesWithCapture, /* useCapture */ true);
```

Reply:
(note: the `removeEventListeners` function is outside of the class (not a private method on a class), so that its name can be minified in 3P)

Reply:
done